src/org/parenthesin/components/webserver.clj

```diff
@@ -1,17 +1,68 @@
 (ns org.parenthesin.components.webserver
-  (:require [com.stuartsierra.component :as component]))
+  (:require [com.stuartsierra.component :as component]
+            [io.pedestal.interceptor.helpers :refer [before]]
+            [io.pedestal.http :as http]
+            [io.pedestal.http.route :as route]
+            [io.pedestal.log :as log]))
+
+(defn- add-system [service]
+  (before (fn [context] (assoc-in context [:request :components] service))))
+
+(defn- system-interceptors 
+  "Extend to service's interceptors to include one to inject the components
+  into the request object"
+  [service service-map]
+  (update-in service-map
+             [::http/interceptors]
+             #(vec (->> % (cons (add-system service))))))
+
+(defn- base-service [routes config]
+  {:env :prod
+   ::http/type :jetty
```

Comment:
that's cool! it seems pedestal has the whole logic to start and deal with the jetty server by itself, right? I did not know that :)

Reply:
https://github.com/pedestal/pedestal/blob/master/jetty/src/io/pedestal/http/jetty.clj